openjdk.test.load/src/test.load/net/adoptopenjdk/stf/MixedLoadTest.java

```diff
@@ -63,16 +80,37 @@ public void execute(StfCoreExtension test) throws StfException {
 				.addProjectToClasspath("openjdk.test.nio")
 				.addProjectToClasspath("openjdk.test.concurrent")
 				.addJarToClasspath(mauveJar)
-				.addJarToClasspath(fileSystemJar)
-				.setAbortIfOutOfMemory(false)
-				.addSuite("mix")
-				.setSuiteThreadCount(cpuCount - 2, 10)	
-				.setSuiteNumTests(totalTests * 200) // Leave 1 cpu for the JIT. i for GC and set min 20			
-				.setSuiteInventory(inventoryFile) 			// Point at the file which lists the tests
-				.setSuiteRandomSelection();		  			// Randomly pick the next test each time
+				.addJarToClasspath(fileSystemJar);
 				
+		if (isTimeBasedLoadTest) { 
+			loadTestInvocation = loadTestInvocation.setTimeLimit(timeLimit);	// If it's a time based test, stop execution after given time duration
+		}
+		loadTestInvocation = loadTestInvocation.setAbortIfOutOfMemory(false)
+				.addSuite("mix")
+				.setSuiteThreadCount(cpuCount - 2, 10); 
+		if (isTimeBasedLoadTest) { 
+			loadTestInvocation = loadTestInvocation
+				.setSuiteNumTests(totalTests * 80000000 * testCountMultiplier); // If it's a time based test, run a very large load
+		} else {  
+			loadTestInvocation = loadTestInvocation
+				.setSuiteNumTests(totalTests * 200);	// Leave 1 cpu for the JIT. i for GC and set min 20
```

Comment:
Comment is wrong (looks like it was wrong before this PR - it should be on the `.setSuiteThreadCount(cpuCount - 2, 10)` line, and should say `min 2` not `min 20`.

Reply:
Yes, this comment was wrong from before. Thanks for catching! Updated.